### PR TITLE
Prepare  Task validation to be used in state tests

### PR DESCRIPTION
**Pull request by tanertopal**
#### What does this implement/fix? Explain your changes.

Preparing  Task validation to be used in state tests

### CODE REVIEW

File: src/py/flwr/server/utils/validator.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+# ==============================================================================
+"""Validators."""
+from typing import List, Union
+
+from flwr.proto.task_pb2 import TaskIns, TaskRes
+
+
+# pylint: disable-next=too-many-branches
+def validate_task_ins_or_res(tasks_ins_res: Union[TaskIns, TaskRes]) -> List[str]:
+    """Validate incoming TaskIns and TaskRes."""
```
Comment (tanertopal) [inline src/py/flwr/server/utils/validator.py:23]:
```suggestion
    """Validate a TaskIns or TaskRes."""
```

File: src/py/flwr/server/utils/validator.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+    # Created/delivered/TTL
+    if tasks_ins_res.task.created_at != "":
+        validation_errors.append("`created_at` must be an empty str")
+    if tasks_ins_res.task.delivered_at != "":
+        validation_errors.append("`delivered_at` must be an empty str")
+    if tasks_ins_res.task.ttl != "":
+        validation_errors.append("`ttl` must be an empty str")
+
+    # TaskInstruction specific
```
Comment (tanertopal) [inline src/py/flwr/server/utils/validator.py:63]:
```suggestion
    # TaskIns specific
```

File: src/py/flwr/server/utils/validator.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                "`task` does not set field `legacy_server_message`"
+            )
+        if not tasks_ins_res.task.legacy_server_message.HasField("msg"):
+            validation_errors.append("`legacy_server_message` does not set field `msg`")
+
+        # Ancestors
+        if len(tasks_ins_res.task.ancestry) != 0:
+            validation_errors.append("`ancestry` is not empty")
+
+    # TaskResult specific
```
Comment (tanertopal) [inline src/py/flwr/server/utils/validator.py:77]:
```suggestion
    # TaskRes specific
```

File: src/py/flwr/server/utils/validator_test.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+            legacy_server_message=ServerMessage(
+                reconnect_ins=ServerMessage.ReconnectIns()
+            ),
+        ),
+    )
+    return task
+
+
+def create_task_res(
+    consumer_node_id: int, anonymous: bool, ancestry: List[str]
```
Comment (danieljanes) [inline src/py/flwr/server/utils/validator_test.py:102]: Should this be `producer_node_id`? For now, the `consumer_node_id` is always expected to be `0`

Comment (tanertopal) [reply src/py/flwr/server/utils/validator_test.py:102]: Fixing it.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**